Added Weaver configuration for all telemetry, and a generated mkdocs site

martinjt opened the pull request:
# Changes

This adds a new service called `telemetry-docs` which serves a documentation site for the custom telemetry documented by the demo across all the services.

It also includes a weaver registry for the all those attributes.

This does not include a live-check interface, that will come later, nor does this include codegen for these attributes or metrics.

## Merge Requirements

For new features contributions, please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

martinjt committed:
Added Weaver configuration for all telemetry, and a generated mkdocs site

julianocosta89 commented:
@martinjt can we configure live-check to ensure the telemetry schema is always up to date?

I just don't want that to be another burden for maintainers.

Also, the k8s manifests shouldn't be part of this PR. That is generated via helm update whenever we have a new release

martinjt commented:
We can, i just didnt want to make it part of this PR since its already got a lot of moving parts.

I can remove the k8s bits.

puckpuck committed:
Merge branch 'main' into weaver-docs

puckpuck committed:
fix redirect

puckpuck committed:
use TELEMETRY_DOCS_PORT

puckpuck committed:
Merge branch 'main' into weaver-docs

puckpuck committed:
add changelog entry

puckpuck committed:
Merge branch 'weaver-docs' of https://github.com/martinjt/opentelemetry-demo into weaver-docs

puckpuck committed:
cleanup lint errors

puckpuck committed:
more lint fixes

puckpuck committed:
add telemetry-docs dependency

julianocosta89 commented:
@martinjt I had a chat with @puckpuck in the last SIG meeting and we were discussing this PR.
As an OTel user, I couldn't understand why would we run the documentation as part of the Demo.

Shouldn't we keep this PR just with the `/telemetry-schema` and have the telemetry docs as part of the https://opentelemetry.io/docs/demo/ docs?

Something like how the semconv docs is currently done.

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 committed:
Move docs to nginx, and cleanup k8s manifests

julianocosta89 committed:
Update README

julianocosta89 commented:
@martinjt I've touched your PR, hope you don't mind.
Here are the things I've changed:

- Removed the k8s manifests (those are updated via Helm only when we have a new release)
- Reduced image size from 236MB to 102MB, by switching from mkdocs serve with Python runtime to static site build + nginx
  - With that we also gained traces from the telemetry docs service.
- Set static Weaver version on the Dockerfile to avoid breaking changes from `latest`.
- Bumped Python version to 3.14

This should be good to go, but as I've changed a couple of things, I'll wait for your review.

julianocosta89 committed:
Fix sanity and markdownlint

julianocosta89 committed:
Add telemetry docs to build images workflow

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 committed:
Merge branch 'main' into weaver-docs

martinjt committed:
Upgrade Weaver to v0.22.1 and migrate schemas to v2 format

- Bump otel/weaver Docker image from v0.21.2 to v0.22.1
- Rename registry_manifest.yaml to manifest.yaml (v0.22 deprecation)
- Migrate attribute files to v2 schema (file_format: definition/2)
- Migrate service files to v2 attribute_groups with visibility
- Update jq filters in weaver.yaml for new resolved registry ID format
- Add Apache license headers to Jinja2 template files
- Fix readme.md.j2 to reflect actual schema file structure
- Add TELEMETRY_DOCS_HOST/PORT env vars to docker-compose.minimal.yml
- Update mkdocs site_url to localhost:8080/telemetry/

Note: Claude was used to assist with this change. The author has
verified all work.

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 committed:
Merge branch 'main' into weaver-docs

julianocosta89 reviewed (approved):
I'd get this one merged and we can iterate from here
